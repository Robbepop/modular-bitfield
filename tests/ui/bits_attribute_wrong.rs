// This is just the compile_fail version of the previous test case, testing what
// error happens if the user has written an incorrect #[bits = N] attribute.
//
// Ensure that the error message points to the incorrect attribute and contains
// the correct number of bits in some form.

use modular_bitfield::prelude::*;

#[bitfield]
pub struct RedirectionTableEntry {
    #[bits = 9]
    trigger_mode: TriggerMode,
    reserved: B7,
}

#[derive(BitfieldSpecifier, Debug)]
pub enum TriggerMode {
    Edge = 0,
    Level = 1,
}

fn main() {}
