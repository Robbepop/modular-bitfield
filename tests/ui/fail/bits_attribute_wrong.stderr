error[E0308]: mismatched types
  --> tests/ui/fail/bits_attribute_wrong.rs:11:7
   |
11 |     #[bits = 9]
   |       ^^^^ expected an array with a size of 9, found one with a size of 1
